### PR TITLE
(#10) Cake build

### LINKED ISSUE #10: (build) use Cake to build the project, with our standard recipe
We want to move to having standard build practices across our repositories. As part of that effort, we should remove the custom build.ps1 we use here and instead have a standard cake.recipe we use to build the project.

This would get us built in test and potentially code coverage reporting, among many other things, and would make it easier to manage this project in our existing CI/CD systems.

### PR DESCRIPTION
**Pull request by vexx32**
## Description Of Changes

- Restructure the project a bit to align with our standards for other projects.
- Implement a Cake build based on our base template.

NOTE: In discussion with @gep13 we realised the current way our template builds nupkgs is not especially suitable for .NET Core builds as it's not using `dotnet pack` to generate the nupkgs. Conclusion there was that it's probably worth updating our template to support that more fully instead of going down the road we have in some other projects where files added to the nupkg are manually specified in the nuspec, since this is a bog-standard NuGet package and not a Chocolatey package.

Until that change has been made to the template recipe we use and the version ref in this cake.recipe is updated, nupkgs will not be build with this cake.recipe script.

## Motivation and Context

Moving towards standard build processes for all our stuff, will be good to have a standard build process for this.

Apologies for the insane diffs, I'm not sure what happened with a lot of these, the files were simply moved. Best guess is Git was remembering file formats (including BOMs) for existing files / they hadn't been modified in a very long time, and as soon as a change was made Git assumed the file format had changed or something. 😔 

## Testing

1. Run `build.ps1` to build the project and run tests.

Note: this project requires the tests to be run from an Administrator session as part of the tests in this project involve setting up service URL bindings. You may see Windows Firewall prompts to allow these tests to do their thing as well.

## Change Types Made

* [x] CI changes
* [ ] Bug fix (non-breaking change)
* [ ] Feature / Enhancement (non-breaking change)
* [ ] Breaking change (fix or feature that could cause existing functionality to change)

## Related Issue

Fixes #10

## Change Checklist

* [ ] Requires a change to the documentation
* [ ] Documentation has been updated
* [ ] Tests to cover my changes, have been added
* [x] All new and existing tests passed.

### CODE REVIEW

File: test/Rhino.Licensing.Tests/Rhino.Licensing.Tests.csproj
Comment (vexx32) [reply test/Rhino.Licensing.Tests/Rhino.Licensing.Tests.csproj:43]: Filed https://github.com/chocolatey/rhino-licensing/issues/11

**Review by corbob — APPROVED** [review]
Looks good to me! Thanks for getting this done @vexx32

**Comment by vexx32** [comment]
@gep13 I've updated the PR per your comments in Slack. 🙂

**Review by corbob — APPROVED** [review]
All the changed files since last review look good to me. Will hold off on the merge until @Windos has had a chance to verify the import with Ansible.

File: .teamcity/settings.kts
Comment (Windos) [inline .teamcity/settings.kts:24]: The root project has `env.NUGETDEV_SOURCE` (and `env.NUGETDEV_API_KEY`) parameters that will be available if you'd prefer to use that/those. If so, remove these lines and change references in the actual code (i.e. `NUGET_DEV_SOURCE` -> `NUGETDEV_SOURCE`)

This isn't a hard requirement, just making sure you're aware.

File: recipe.cake
Comment (Windos) [inline recipe.cake:18]: Just flagging these as possibly being change to `NUGETDEV_SOURCE` if you decide to go that route.

**Review by Windos — CHANGES_REQUESTED** [review]

File: .teamcity/settings.kts
Comment (vexx32) [reply .teamcity/settings.kts:28]: The main branches this will be running on would be master I think at least for now. PR branches maybe in the future potentially?

Comment (Windos) [reply .teamcity/settings.kts:28]: No need to change line 27, the complete section ends up as:

```
    vcs {
        root(DslContext.settingsRoot)
    }
```

File: recipe.cake
Comment (vexx32) [reply recipe.cake:18]: think I'd need to remove one of them or release builds would fail since it tries to push to both on release builds as I understand it, but yep, thanks!

File: .teamcity/settings.kts
Comment (Windos) [inline .teamcity/settings.kts:9]: No action to take here just yet (will do a quick test), but this line _can_ be removed (but not sure if it _needs_ to be removed or not) will report back.

File: .teamcity/settings.kts
Comment (Windos) [inline .teamcity/settings.kts:21]:
```suggestion
        param("env.Git_Branch", "%teamcity.build.vcs.branch.RhinoLicensingAnsible_RhinoAnsibleVcsRoot%")
```

This line was erroring on the import now, and this is the format used by other builds using `env.Git_Branch`

**Review by Windos — CHANGES_REQUESTED** [review]

**Review by Windos — APPROVED** [review]
Looks good to me, importing into TeamCity without issue.

**Comment by gep13** [comment]
@vexx32 is there a reason that we didn't switch to the new Chocolatey.Cake.Recipe package in this PR?

**Comment by vexx32** [comment]
It was my intention to, but I had forgotten I'd not swapped it over before we merged. I'll do a follow up PR to fix that.

**Pull request merged**
